### ISSUE TITLE
Bust up Write-GitStatus into smaller functions to make it easier for individuals to override parts of the status display

### ISSUE THREAD

**Issue by rkeithhill** [dahlbyk/posh-git#345]
I see a number of PRs come across to change the Git status summary display for things that probably don't apply to that many folks.  And folks can today write their own prompt function to take the information from `Get-GitStatus` to display the Git status summary, plus any additional information in anyway they would like.  However, in many cases folks just want to tweak one area of the Git status summary info.  With this in mind, we should break up the bits and pieces of `Write-GitStatus` into reusable functions.  

Perhaps we have `Write-GitStatus` call down to `Write-BranchStatus`, `Write-GitIndexStatus`, and `Write-GitWorkingDirStatus`. 

In one PR scenario the OP wants to display any tags associated with HEAD.  In this case, the OP could create his own version of Write-BranchStatus to display tags. Then in his prompt function he could call `Write-GitIndexStatus` and `Write-GitWorkingDirStatus` to create the rest of the status summary text. This would make overriding an individual status summary section much easier.

You could make the functions even finer grained as well. Perhaps break down `Write-BranchStatus` into `Write-BranchName` and `Write-BranchSyncStatus`?

**Comment by dahlbyk** [comment]
This is also related to #304 and https://github.com/Jaykul/PowerLine/issues/1. More modularity and customizability in how the prompt displays would be **awesome**. And if we do it right, it seems like an opportunity to remove a bunch of clutter from `$GitPromptSettings`, e.g. default might be:

```powershell
function Write-GitStatus ($GitStatus) {
  Write-Prompt ' [' -ForegroundColor Yellow # No need for Before settings
  Write-GitBranchStatus $GitStatus
  Write-GitFileStatus $GitStatus
  Write-GitLocalStatus $GitStatus
  Write-Prompt ']' -ForegroundColor Yellow # No need for After settings
}
```

And to customize:

```powershell
function Write-GitStatus ($GitStatus) {
  Write-Prompt ' {'
  Write-GitBranchStatus $GitStatus -AheadBehindDisplay Compact -GoneText ''
  # No need for EnableFileStatus; just don't call Write-GitFileStatus
  Write-GitLocalStatus -StagedText ''
  Write-GitStashStatus -Before ' (' -After '(' -ForegroundColor Red
  Write-Prompt '}'
}
```

**Comment by rkeithhill** [comment]
I've implemented this on branch - https://github.com/dahlbyk/posh-git/tree/rkeithhill/refactor-write-prompt  This branch is a branch off of the `settings-only-update` branch.

**Comment by rkeithhill** [comment]
Fixed via PR #513
